src/client/particles.cpp

```diff
@@ -75,36 +78,30 @@ Particle::Particle(
 	// Set material
 	{
 		// translate blend modes to GL blend functions
-		video::E_BLEND_FACTOR bfsrc, bfdst;
-		video::E_BLEND_OPERATION blendop;
+		bool blend = true;
+		video::E_BLEND_FACTOR bfsrc = video::EBF_SRC_ALPHA, bfdst = video::EBF_DST_ALPHA;
+		video::E_BLEND_OPERATION blendop = video::EBO_ADD;
 		const auto blendmode = texture.tex != nullptr
 				? texture.tex->blendmode
 				: ParticleParamTypes::BlendMode::alpha;
 
 		switch (blendmode) {
-			case ParticleParamTypes::BlendMode::add:
-				bfsrc = video::EBF_SRC_ALPHA;
-				bfdst = video::EBF_DST_ALPHA;
-				blendop = video::EBO_ADD;
-			break;
-
+			case ParticleParamTypes::BlendMode::clip:
+				blend = false;
+				break;
+			case ParticleParamTypes::BlendMode::alpha:
+				bfdst = video::EBF_ONE_MINUS_SRC_ALPHA;
+				break;
+			case ParticleParamTypes::BlendMode::add: break;
 			case ParticleParamTypes::BlendMode::sub:
-				bfsrc = video::EBF_SRC_ALPHA;
-				bfdst = video::EBF_DST_ALPHA;
 				blendop = video::EBO_REVSUBTRACT;
-			break;
-
+				break;
 			case ParticleParamTypes::BlendMode::screen:
 				bfsrc = video::EBF_ONE;
 				bfdst = video::EBF_ONE_MINUS_SRC_COLOR;
-				blendop = video::EBO_ADD;
-			break;
-
-			default: // includes ParticleParamTypes::BlendMode::alpha
-				bfsrc = video::EBF_SRC_ALPHA;
-				bfdst = video::EBF_ONE_MINUS_SRC_ALPHA;
-				blendop = video::EBO_ADD;
-			break;
+				break;
+			case ParticleParamTypes::BlendMode::BlendMode_END:
+				throw std::logic_error("invalid blend mode");
```

Comment:
1. you forgot to handle all other invalid enum values
2. i'm a bit wondering if there's any point. the value is checked on deseriailization and it's not like any code is going to catch this and maybe do something with it. it will always be a fatal error. and it's potential problems for future changes

Reply:
The point is basically to avoid a compiler warning by handling all cases; 
I do want a hypotethical compiler warning if someone adds a new blend mode and forgets to update this (hence I'm not just using `default`). I should maybe explain this in a comment. 
I indeed don't care about other invalid values, because I validate on deserialization.

> and it's potential problems for future changes

How so?